packages/server/src/worker/workerTypes.ts

```diff
@@ -68,7 +68,8 @@ export interface TJobSyncAlgoliaData {
 export interface TJobSyncAlgoliaResult {
   kind: JobKind.SyncAlgolia;
   jobName: string;
-  wasUpdated: boolean;
+  wasUpdated: boolean; // true if the index was successfully updated; false otherwise
```

Comment:
I probably wouldn't use `wasUpdated` here. It takes a comment to explain and I feel like `succeeded` would be more than enough for readers to understand.

There's another issue here -- is it the case that if `wasUpdated` is `false` the `errorMessage` will. be populated, and visa versa? If that's true then we're using two fields to tell us one thing. Also, I kinda led you down the wrong path here I think. Perhaps we should just throw if `index.saveObjects` or `index.deleteObjects` fails. We'll see the failure in the worker queue for sure... but on the other hand if we catch it and save as it is now that could give us a place to retry down the road.

If it's not obvious -- API design (in this case designing our in/out types) is one of the most tricky bits to this kind of software engineering. A lot of it is an art and a lot of it you have to learn from experience (yours or someone else's). APIs are also one of the hardest things to change once they are in the wild since they are necessarily the boundary of a system and so other systems are depending on them. My team goes back and forth forever on this part; the actual implementation normally is pretty easy once we've put a lot of thought into the interface.

Reply:
I'd prob just throw the algolia error. Sorry your PRs have been subject to the whims of my thought process.

Reply:
And I'd probably put `deletedRecords: string[] // ObjectIds` and `insertedRecords: string[]` (are they inserted or updated?) in the response.

Reply:
They are inserted and updated depending on if Algolia has the id already :)